### PR TITLE
Add JSON output for single file classification

### PR DESCRIPTION
**Pull request by lbonanomi**
This PR attempts to resolve https://github.com/github/linguist/issues/3562 by outputting JSON-format results for a single file.

## Description

Example based on issue:

```
$ ./bin/github-linguist grammars.yml --json
{"grammars.yml":{"lines":857,"sloc":856,"type":"Text","mime_type":"text/x-yaml","language":"YAML"}}
```

- [X] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

### CODE REVIEW

File: bin/github-linguist [hunk 1]
```diff
@@ -56,10 +57,21 @@ elsif File.file?(path)
...
+  else
+    puts JSON.generate( { blob.name =>	{
+	    					:lines => blob.loc,
+						:sloc => blob.sloc,
+						:type => type,
+						:mime_type => blob.mime_type,
+						:language => blob.language
+    					}
+    			} )
+  end
```
Comment (lildude) [inline bin/github-linguist]: Negation in the first part of an `if` statement can lead to confusion. I know we have this smelly behaviour elsewhere in this file and we can address that at a later date. For the moment, please flip this around, ie:

```ruby
if json_breakdown
  puts JSON.generate( { blob.name =>	{
  [...]
else
  puts "#{blob.name}: #{blob.loc} lines (#{blob.sloc} sloc)"
  [...]
end

File: bin/github-linguist [hunk 2]
```diff
@@ -22,10 +22,10 @@ end
 ARGV.shift
 breakdown = true if ARGV[0] == "--breakdown"
 json_breakdown = true if ARGV[0] == "--json"
-
```
Comment (lildude) [inline bin/github-linguist]: Please put this blank line back. It makes things clearer.

File: bin/github-linguist (hunk 1 above)
Comment (lbonanomi) [reply bin/github-linguist]: I'll refrain from making a joke about perl, here. Done.

File: bin/github-linguist (hunk 2 above)
Comment (lbonanomi) [reply bin/github-linguist]: You're right, it is re-added.

**Pull request merged**